spanner/opencensus/src/main/java/com/example/spanner/opencensus/CaptureGfeMetric.java

```diff
@@ -0,0 +1,175 @@
+    captureGfeMetric(dbClient);
+  }
+
+  // [START spanner_opencensus_capture_gfe_metric]
+  private static final String MILLISECOND = "ms";
+  private static final TagKey key = TagKey.create("grpc_client_method");
+
+  // GFE t4t7 latency extracted from server-timing header.
+  public static final MeasureLong SPANNER_GFE_LATENCY =
+      MeasureLong.create(
+          "cloud.google.com/java/spanner/gfe_latency",
+          "Latency between Google's network receives an RPC and reads back the first byte of the"
+              + " response",
+          MILLISECOND);
+
+  static final Aggregation AGGREGATION_WITH_MILLIS_HISTOGRAM =
+      Distribution.create(BucketBoundaries.create(Arrays.asList(
+          0.0, 0.01, 0.05, 0.1, 0.3, 0.6, 0.8, 1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 8.0, 10.0, 13.0,
+          16.0, 20.0, 25.0, 30.0, 40.0, 50.0, 65.0, 80.0, 100.0, 130.0, 160.0, 200.0, 250.0,
+          300.0, 400.0, 500.0, 650.0, 800.0, 1000.0, 2000.0, 5000.0, 10000.0, 20000.0, 50000.0,
+          100000.0)));
+  static final View GFE_LATENCY_VIEW = View
+      .create(Name.create("cloud.google.com/java/spanner/gfe_latency"),
+          "Latency between Google's network receives an RPC and reads back the first byte of the"
+              + " response",
+          SPANNER_GFE_LATENCY,
+          AGGREGATION_WITH_MILLIS_HISTOGRAM,
+          Collections.singletonList(key));
+
+  static ViewManager manager = Stats.getViewManager();
+
+  private static final Tagger tagger = Tags.getTagger();
+  private static final StatsRecorder STATS_RECORDER = Stats.getStatsRecorder();
+
+  static void captureGfeMetric(DatabaseClient dbClient) {
+    // Register GFE view.
+    manager.registerView(GFE_LATENCY_VIEW);
+
+    // Enable OpenCensus exporters to export metrics to Stackdriver Monitoring.
+    // Exporters use Application Default Credentials to authenticate.
+    // See https://developers.google.com/identity/protocols/application-default-credentials
+    // for more details.
+    try {
+      StackdriverStatsExporter.createAndRegister();
+    } catch (IOException | IllegalStateException e) {
+      System.out.println("Error during StackdriverStatsExporter");
+    }
+
+    try (ResultSet resultSet =
+        dbClient
+            .singleUse() // Execute a single read or query against Cloud Spanner.
+            .executeQuery(Statement.of("SELECT SingerId, AlbumId, AlbumTitle FROM Albums"))) {
+      while (resultSet.next()) {
+        System.out.printf(
+            "%d %d %s", resultSet.getLong(0), resultSet.getLong(1), resultSet.getString(2));
+      }
+    }
+  }
+
+  private static final HeaderClientInterceptor interceptor = new HeaderClientInterceptor();
+  private static final Metadata.Key<String> SERVER_TIMING_HEADER_KEY =
+      Metadata.Key.of("server-timing", Metadata.ASCII_STRING_MARSHALLER);
+  private static final Pattern SERVER_TIMING_HEADER_PATTERN = Pattern.compile(".*dur=(?<dur>\\d+)");
```

Comment:
OPTIONAL - consider mentioning where this came from.

Reply:
Done. Thanks